### PR TITLE
Loading activity

### PR DESCRIPTION
**Pull request by 3lvis**

### CODE REVIEW

File: Source/ViewableController.swift [hunk 1]
```diff
@@ -280,7 +289,17 @@ class ViewableController: UIViewController {
...
+            // loading indicator will only be displayed when the image is taking a lot of time to load.
+            let deadline = DispatchTime.now() + Double(Int64(0.5 * Double(NSEC_PER_SEC))) / Double(NSEC_PER_SEC)
+            DispatchQueue.main.asyncAfter(deadline: deadline) {
+                if self.imageView.image == nil {
+                    self.imageLoadingIndicator.startAnimating()
+                }
+            }
+
             viewable.media { image, _ in
+                self.imageLoadingIndicator.stopAnimating()
```
Comment (designatednerd) [inline Source/ViewableController.swift]: Am I correct in my assumption that this will always get called back on the main thread?

Comment (3lvis) [reply Source/ViewableController.swift]: Will be called in the main thread :)

Comment (3lvis) [reply Source/ViewableController.swift]: This is a protocol that any Viewable should implement, so it's the responsibility of the caller to return this on the main thread. I could always add an extra check, though.

**Comment by 3lvis** [comment]
Should be good to go, @designatednerd  :)

**Review by designatednerd — APPROVED** [review]

**Comment by 3lvis** [comment]
Thanks @Amefuri and @designatednerd!

**Pull request merged**